lib/middleware.js

```diff
@@ -512,6 +516,22 @@ module.exports = function middlewareConstructor(env) {
           res.json({"status": 200});
         });
       };
+    },
+
+    /**
+     * If there's an oauth token, decrypt it and set the
+     * local user object.
+     */
+    decryptToken: function(req, res, next) {
+      if (!req.session.token) {
```

Comment:
but what happens if it doesn't anyway? (if the error handler catches that, good enough)